serverless/virus-scanner/src/index.ts

```diff
@@ -0,0 +1,145 @@
+import { getLambdaLogger } from './logger'
+import { S3Service } from './s3.service'
+import { isBodyWithKey } from './types'
+
+/**
+ * Handler for virus scanner lambda
+ * To invoke handle locally, send POST request to http://localhost:9999/2015-03-31/functions/function/invocations with the following request body:
+ * { "key": <object key>}
+ * @param event
+ * @returns 200 with clean file key if clean
+ * @returns 400 if malicious
+ * @returns 400 if body is missing key
+ * @returns 404 if file not found
+ */
+export const handler = async (
+  event: unknown,
+): Promise<APIGatewayProxyResult> => {
+  const logger = getLambdaLogger('virus-scanner')
+
+  logger.info('handler triggered')
+
+  if (!isBodyWithKey(event)) {
+    logger.warn({
+      message: 'Missing key in body',
+      event,
+    })
+    return {
+      statusCode: StatusCodes.BAD_REQUEST,
+      body: JSON.stringify({
+        message: 'Missing key in body',
+      }),
+    }
+  }
+
+  if (!validate(event.key)) {
+    logger.warn({
+      message: 'Invalid key',
+      event,
+    })
+    return {
+      statusCode: StatusCodes.BAD_REQUEST,
+      body: JSON.stringify({
+        message: 'Invalid key',
+      }),
+    }
+  }
+
+  const quarantineFileKey = event.key
+
+  const cleanBucket = config.virusScannerCleanS3Bucket
+  const quarantineBucket = config.virusScannerQuarantineS3Bucket
+
+  // Retrieve from S3
+  const s3Client = new S3Service(config.isTestOrDev, logger)
+
+  let s3ReadableStream
+  try {
+    s3ReadableStream = await s3Client.getS3FileStreamWithVersionId({
+      bucketName: quarantineBucket,
+      objectKey: quarantineFileKey,
+    })
+  } catch (error) {
+    logger.error({
+      message: 'File not found',
+      error,
+      quarantineFileKey,
+    })
+    return {
+      statusCode: StatusCodes.NOT_FOUND,
+      body: JSON.stringify({
+        message: 'File not found',
+        fileKey: quarantineFileKey,
+      }),
+    }
+  }
+
+  // Scan file
+
+  const { body, versionId } = s3ReadableStream
+
+  const scanResult = await scanFileStream(body)
+  const { isMalicious } = scanResult
+
+  // If malicious, log and delete
+  if (isMalicious) {
+    const { virusMetadata } = scanResult
+
+    logger.error({
+      message: 'malicious file detected!!',
+      virusMetadata,
+      key: quarantineFileKey,
+    })
+
+    // Delete from quarantine bucket
+    await s3Client.deleteS3File({
+      bucketName: quarantineBucket,
+      objectKey: quarantineFileKey,
+    })
+
+    return {
+      statusCode: StatusCodes.BAD_REQUEST,
+      body: JSON.stringify({
+        message: 'Malicious file detected',
+        key: quarantineFileKey,
+        virusMetadata,
+      }),
+    }
+    // If clean, move to clean bucket with randomised key and return
+  } else {
+    const cleanFileKey = crypto.randomUUID()
+    logger.info({
+      message: 'clean file detected',
+      key: quarantineFileKey,
+      versionId,
+    })
+
+    await s3Client.moveS3File({
```

Comment:
Should be wrapped in a try catch

Reply:
[87e9e45](https://github.com/opengovsg/FormSG/pull/6642/commits/87e9e4560d9391c071234e53b9328165ea941e33)